services/network/SerialServer.cpp

```diff
@@ -12,6 +12,7 @@ namespace services
     {
         serialCommunication.ReceiveData([this](infra::ConstByteRange buffer)
             {
+                really_assert(!receiveQueue.Full());
```

Comment:
I would rather see a really_assert in QueueForOneReaderOneIrqWriter<T>::AddFromInterrupt, and that it does not just check for Full() but whether buffer.size() fits